@here/harp-map-controls/lib/MapControls.ts

```diff
@@ -463,21 +475,43 @@ export class MapControls extends THREE.EventDispatcher {
         }
 
         this.dispatchEvent(MAPCONTROL_EVENT_BEGIN_INTERACTION);
-
         // Register the zoom request
         this.m_startZoom = this.currentZoom;
-        this.m_zoomDeltaRequested = zoomLevel - this.zoomLevelTargetted;
-
+        this.m_zoomDeltaRequested = zoomLevel - this.zoomLevelTargeted;
         // Cancel panning so the point of origin of the zoom is maintained.
         this.m_panDistanceFrameDelta.set(0, 0, 0);
         this.m_lastAveragedPanDistance = 0;
 
-        this.m_zoomTargetNormalizedCoordinates.set(screenTarget.x, screenTarget.y);
-
         // Assign the new animation start time.
         this.m_zoomAnimationStartTime = performance.now();
 
-        this.handleZoom();
+        if (this.mapView.projection.type === geoUtils.ProjectionType.Planar) {
+            this.m_zoomTargetNormalizedCoordinates.set(screenTarget.x, screenTarget.y);
+            this.handleZoom();
+        } else {
+            const surfaceNormal = this.mapView.projection.surfaceNormal(
```

Comment:
Shouldn't all this code go to handleZoom? What do we do differently when zooming in on the globe? Do we have a zoom animation for globe?

Reply:
This PR only integrates the existing globe controls. I want to create further tickets to improve their handling in MapView and have a consistent approach, but at the moment it cannot easily be. With the planar version we zoom by specifying a zoom level in this method, not yet in globecontrols, I have to figure that out. In the planar version the panning is actually a rotation. The orbitting does not have the same reference and up axis. All that calls for different implementation that I want to break down in tickets. The progress will be clearer, more visible for the team, and easier to review.